bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -759,7 +759,7 @@ protected void extractMetaFromEntryLogs() throws EntryLogMetadataMapException {
                 } else {
                     entryLogMetaMap.put(entryLogId, entryLogMeta);
                 }
-            } catch (IOException e) {
+            } catch (IOException | RuntimeException e) {
```

Comment:
what if the RuntimeException is thrown for a different reason and can't be that easily skipped
I think it makes more sense to make EntryLogger.getEntryLogMetadata throw the IOException in this case

Reply:
This try-catch code block reads each entry log file's metadata and checks if this entry log file can be deleted.  The RuntimeException will be thrown by the read entry log file's metadata logic. We should block the whole garbage collection thread if one entry log file's metadata read failed. We can skip this entry log file if it is broken.